### PR TITLE
add secrets to cloud func 2

### PR DESCRIPTION
**Pull request by DrFaust92**
Closes https://github.com/hashicorp/terraform-provider-google/issues/12345

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
cloudfunctions2: add `secret_environment_variables` and `secret_volumes` to `google_cloudfunctions2_function`
```

### CODE REVIEW

File: mmv1/templates/terraform/examples/cloudfunctions2_secret_env.tf.erb [hunk 1]
```diff
@@ -0,0 +1,67 @@
+# [START functions_v2_basic]
```
Comment (c2thorn) [inline mmv1/templates/terraform/examples/cloudfunctions2_secret_env.tf.erb:1]:
```suggestion
```
Mind deleting the lines starting with `# [` (looks like there are 4 introduced in the PR) ? They serve a different purpose and are not needed in these examples

**Review by c2thorn — CHANGES_REQUESTED** [review]
Hit permission errors for the service account (needed `roles/secretmanager.secretAccessor`). I'll rerun the tests, but in the meantime I have a small request

File: mmv1/templates/terraform/examples/cloudfunctions2_secret_env.tf.erb (hunk 1 above)
Comment (DrFaust92) [reply mmv1/templates/terraform/examples/cloudfunctions2_secret_env.tf.erb]: Done

**Review by c2thorn — APPROVED** [review]
Tests are green, thanks @DrFaust92

**Pull request merged**